source/extensions/upstreams/http/config.cc

```diff
@@ -111,7 +111,11 @@ ProtocolOptionsConfigImpl::ProtocolOptionsConfigImpl(
     use_http2_ = true;
     use_alpn_ = true;
     use_http3_ = options.auto_config().has_http3_protocol_options();
-    if (options.auto_config().has_alternate_protocols_cache_options()) {
+    if (use_http3_) {
+      if (!options.auto_config().has_alternate_protocols_cache_options()) {
+        throw EnvoyException(fmt::format("alternate protocols cache must be configured when HTTP/3 "
```

Comment:
lazy q, we have a test for this?

Reply:
Apparently not! https://storage.googleapis.com/envoy-pr/3ec605e/coverage/source/extensions/upstreams/http/config.cc.gcov.html :) Time to learn about testing exceptions!

Done!

Reply:
EXPECT_THROW and friends :-)